lib/project.ts

```diff
@@ -392,12 +388,17 @@ export class Project implements Project.IProject {
 			if (normalizedPropertyName === this.$projectConstants.CORE_PLUGINS_PROPERTY_NAME) {
 				this.$projectPropertiesService.updateCorePlugins(this.projectData, this.configurationSpecificData, mode, propertyValues, this.getConfigurationsSpecifiedByUser()).wait();
 			} else {
+				if (projectConfigurations.length) {
+					_.each(projectConfigurations, configuration => {
+						this.$projectPropertiesService.updateProjectProperty(this.projectData, this.configurationSpecificData[configuration], mode, normalizedPropertyName, propertyValues).wait();
+					});
+				} else {
 				this.$projectPropertiesService.updateProjectProperty(this.projectData, undefined, mode, normalizedPropertyName, propertyValues).wait();
-				_.each(this.configurationSpecificData, configSpecificData => this.$projectPropertiesService.removeProjectProperty(configSpecificData, normalizedPropertyName, this.projectData));
+			}
 			}
 
 			this.saveProject(this.getProjectDir().wait(), projectConfigurations).wait();
-			this.printProjectProperty(normalizedPropertyName).wait();
+			this.printProjectProperty(normalizedPropertyName, projectConfigurations[0]).wait();
```

Comment:
I think it will be better if in the message for the updated property we include all the project configurations that are updated because if the user wants to change the Author property for multiple project configurations and the message after the command execution is `The value of Author is: SomeAuthor` it can be confusing - is it changed only in .abproject or in all project configuration files or is it changed for the specified configurations.